File: src/main/java/seremis/geninfusion/soul/gene/newAI/swimming/GeneAISwimming.java
package seremis.geninfusion.soul.gene.newAI.swimming;

import seremis.geninfusion.api.soul.IAllele;
import seremis.geninfusion.api.soul.lib.Genes;
import seremis.geninfusion.soul.MasterGene;
import seremis.geninfusion.soul.allele.AlleleBoolean;

public class GeneAISwimming extends MasterGene {

    public GeneAISwimming() {
        addControlledGene(Genes.GENE_AI_SWIMMING);
    }

    @Override
    public Class<? extends IAllele> possibleAlleles() {
        return AlleleBoolean.class;
    }
}

File: src/main/scala/seremis/geninfusion/soul/gene/newAI/GeneAISwimming.scala
package seremis.geninfusion.soul.gene.newAI

import seremis.geninfusion.api.lib.Genes
import seremis.geninfusion.soul.MasterGene

class GeneAISwimming extends MasterGene(classOf[Boolean]) {
    addControlledGene(Genes.GeneAISwimming)
}
